### PR TITLE
add gitlab support

### PR DESCRIPTION
**Pull request by bumblebee89**
Signed-off-by: Bammeke Michael <mike_bammeke@yahoo.com>

### CODE REVIEW

File: sources/cmd/syncdatasources/syncdatasources.go [hunk 1]
```diff
@@ -140,6 +141,10 @@ var (
 			"DA_PIPERMAIL_HTTP_TIMEOUT":   "60s",
 			"DA_PIPERMAIL_NO_INCREMENTAL": "1",
 		},
+		lib.Gitlab: {
+			"DA_PIPERMAIL_HTTP_TIMEOUT":   "60s",
```
Comment (lukaszgryglicki) [inline sources/cmd/syncdatasources/syncdatasources.go:145]: why DA_PIPERMAIL in `GitLab` ds? Shouldn't it be `DA_GITLAB_HTTP_TIMEOUT`?

Comment (bumblebee89) [reply sources/cmd/syncdatasources/syncdatasources.go]: @lukaszgryglicki fixed. Nice catch!... Thank you

Comment (lukaszgryglicki) [reply sources/cmd/syncdatasources/syncdatasources.go]: Of course it depends on how your `da-ds` implementation reads `env` config - but see other data sources for example search 9case insensitive) `rocket` in this SDS source file...

File: sources/gitlab.go [hunk 2]
```diff
@@ -0,0 +1,173 @@
+package syncdatasources
```
[the remark this answers is not in the document]
Comment (bumblebee89) [reply sources/gitlab.go:1]: Done

**Review by lukaszgryglicki — APPROVED** [review]
/lgtm, but please see my comments and also please take `sources/cmd/syncdatasources/syncdatasources.go` and search (case insensitive) for other data sources (like for example `rocket`) and include similar code for `gitlab` everywhere where `rocket` code is referenced, especially I would also add in:
- `func figureOutEndpoints`: `lib.Git, lib.Jira, lib.Bugzilla, lib.BugzillaRest, lib.Jenkins, lib.Gerrit, lib.Pipermail, lib.Confluence, lib.GitHub, lib.Discourse, lib.RocketChat` -> `lib.Git, lib.Jira, lib.Bugzilla, lib.BugzillaRest, lib.Jenkins, lib.Gerrit, lib.Pipermail, lib.Confluence, lib.GitHub, lib.Discourse, lib.RocketChat, lib.Gitlab`.
- `func figureOutDatasourceFromIndexName`: 
```
    lib.MeetUp,
    lib.RocketChat,
  }
```
->
```
    lib.MeetUp,
    lib.RocketChat,
    lib.Gitlab,
  }
```
- `func massageConfig` - maybe similar to GitHub?
- consider `func mapOrigin` - probably not needed... just to confirm
-

**Comment by lukaszgryglicki** [comment]
It's approved, you can merge anytime - ing me when done, so I'll rebuild SDS.

**Comment by lukaszgryglicki** [comment]
This is approved but not merged ... @bumblebee89 ? It becomes outdated/stalled...

**Comment by bumblebee89** [comment]
@lukaszgryglicki going ahead to merge this. Will ping when we are ready for a build. Thanks.

**Comment by lukaszgryglicki** [comment]
Sure

**Pull request merged**